libxw/src/libxw/puzzle.py

```diff
@@ -0,0 +1,49 @@
+from dataclasses import dataclass
+from enum import Enum
+from enum import auto
+from typing import Mapping
+from typing import Sequence
+from typing import Tuple
+from typing import TypeAlias
+from typing import Union
+
+
+CellIndex: TypeAlias = Tuple[int, int]
+
+
+class CellBackground(Enum):
+    WHITE = auto()
+    GRAY = auto()
+
+
+@dataclass
+class Cell:
+    chars: Sequence[str]  # Unless this is a rebus, this should be a singleton list.
+    is_circled: bool  # Some puzzles have letters inside of circles like ⓣⓗⓘⓢ.
+    background: CellBackground
+
+    @property
+    def is_rebus(self) -> bool:
+        return len(self.chars) > 1
+
+
+class ClueDirection(Enum):
+    ACROSS = auto()
+    DOWN = auto()
+
+
+@dataclass
+class Clue:
+    text: str
+
+
+class BlackSquare:
+    pass
```

Comment:
This is a singleton class.. we're just defining it as a class (rather than only a constant) so that the type system can know about it.